wallet/src/wallet/mod.rs

```diff
@@ -257,22 +253,57 @@ impl<B: storage::Backend> Wallet<B> {
         })
     }
 
+    pub fn get_new_public_key(&mut self, account_index: U31) -> WalletResult<PublicKey> {
+        self.for_account_rw(account_index, |account, db_tx| {
+            account.get_new_public_key(db_tx, KeyPurpose::ReceiveFunds)
+        })
+    }
+
+    pub fn get_vrf_public_key(&mut self, account_index: U31) -> WalletResult<VRFPublicKey> {
+        self.accounts
+            .get(&account_index)
+            .ok_or(WalletError::NoAccountFoundWithIndex(account_index))?
+            .get_vrf_public_key()
+    }
+
     pub fn create_transaction_to_addresses(
         &mut self,
         account_index: U31,
         outputs: impl IntoIterator<Item = TxOutput>,
     ) -> WalletResult<SignedTransaction> {
         let request = SendRequest::new().with_outputs(outputs);
-        let tx = self.for_account_rw(account_index, |account, db_tx| {
+        self.for_account_rw(account_index, |account, db_tx| {
             account.process_send_request(db_tx, request)
-        })?;
-        Ok(tx)
+        })
+    }
+
+    pub fn create_stake_pool_tx(
+        &mut self,
+        account_index: U31,
+        amount: Amount,
+    ) -> WalletResult<SignedTransaction> {
+        self.for_account_rw(account_index, |account, db_tx| {
+            account.create_stake_pool_tx(db_tx, amount)
+        })
+    }
+
+    pub fn get_pos_gen_block_data(
+        &mut self,
+        account_index: U31,
+    ) -> WalletResult<PoSGenerateBlockInputData> {
+        let account = self
+            .accounts
+            .get(&account_index)
+            .ok_or(WalletError::NoAccountFoundWithIndex(account_index))?;
+        account.get_pos_gen_block_data()
     }
 
     /// Returns the last scanned block hash and height.
     /// Returns genesis block when the wallet is just created.
     pub fn get_best_block(&self) -> WalletResult<(Id<GenBlock>, BlockHeight)> {
-        Ok((self.best_block_id, self.best_block_height))
+        // TODO: Scan all accounts
+        let account = self.accounts.values().next().ok_or(WalletError::WalletNotInitialized)?;
+        Ok(account.best_block())
```

Comment:
why was best block moved to per account, can they be different?

Reply:
Yes, I decided that it would be better to keep it separate for different accounts. For example, the user can create a new account at a later time, and while the wallet is scanning the node, it should be possible to continue working with the old accounts. The wallet can be stopped at any time and the sync progress should be kept separate.
But this is for a the future, it's not currently implemented.

Reply:
maybe I am not understanding it correctly, wouldn't creating a new account mean there are no transactions associated with it, so nothing to sync?
Or do you mean importing an existing account from a different root key?

Reply:
We can't tell if an account has already been used on another node/wallet or not, and should scan all blocks starting at genesis. The root key can be the same, but the derivation path will be different.

Reply:
thanks, didn't think about that.